addons/binding/org.openhab.binding.xmppclient/src/main/java/org/openhab/binding/xmppclient/internal/XMPPClient.java

```diff
@@ -0,0 +1,117 @@
+import org.jxmpp.stringprep.XmppStringprepException;
+import org.openhab.binding.xmppclient.handler.PublishTriggerChannel;
+import org.openhab.binding.xmppclient.handler.XMPPClientMessageSubscriber;
+
+import java.io.IOException;
+import java.util.HashSet;
+import java.util.Set;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link XMPPClient} is lib foe handle XMPP connection and messagingю
+ *
+ * @author Pavel Gololobov - Initial contribution
+ */
+public class XMPPClient implements IncomingChatMessageListener {
+
+    private final Logger logger = LoggerFactory.getLogger(XMPPClient.class);
+
+    private AbstractXMPPConnection connection;
+
+    private Set<XMPPClientMessageSubscriber> subscribers = new HashSet<>();
+
+    public void subscribe(XMPPClientMessageSubscriber channel) {
+        logger.info("Channel {} subscribed", channel.getName());
+        subscribers.add(channel);
+    }
+
+    public void unsubscribe(XMPPClientMessageSubscriber channel) {
+        logger.info("Channel {} unsubscribed", channel.getName());
+        subscribers.remove(channel);
+    }
+
+    public void connect(String host, Integer port, String login, String password, String resource) {
+        disconnect();
+        try {
+            XMPPTCPConnectionConfiguration config = XMPPTCPConnectionConfiguration.builder()
+                    .setUsernameAndPassword(login, password)
+                    .setXmppDomain(host)
+                    .setResource(resource)
+                    .setPort(port)
+                    .build();
+
+            connection = new XMPPTCPConnection(config);
+
+            ReconnectionManager reconnectionManager = ReconnectionManager.getInstanceFor(connection);
+            reconnectionManager.enableAutomaticReconnection();
+
+            ReconnectionManager.setEnabledPerDefault(true);
+
+            connection.connect().login();
+
+            ChatManager chatManager = ChatManager.getInstanceFor(connection);
+            chatManager.addIncomingListener(this);
+        } catch (SmackException | IOException | XMPPException | InterruptedException e) {
+            logger.error("XMPP CONNECTION ERROR", e);
+        }
+    }
+
+    public void disconnect() {
+        if(connection != null) {
+            connection.disconnect();
+        }
+    }
+
+    public void sendMessage(String to, String message) {
+        if(connection == null) {
+            logger.error("XMPP CONNECTION IS NULL");
+            return;
+        }
+        try {
+            ChatManager chatManager = ChatManager.getInstanceFor(connection);
+            chatManager.addIncomingListener(this);
+
+            EntityBareJid jid = JidCreate.entityBareFrom(to);
+            Chat chat = chatManager.chatWith(jid);
+            chat.send(message);
+        } catch (XmppStringprepException | SmackException.NotConnectedException | InterruptedException e) {
+            logger.error("XMPP SENDING ERROR", e);
```

Comment:
logger. And in your message: Don't shout at your innocence user. You can't blame him for this to have happened.